tools/fast/series/wait_for_chain_message.go

```diff
@@ -0,0 +1,79 @@
+package series
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/filecoin-project/go-filecoin/tools/fast"
+	"github.com/filecoin-project/go-filecoin/types"
+	"github.com/ipfs/go-cid"
+)
+
+// MsgInfo contains the BlockCid for the message MsgCid
+type MsgInfo struct {
+	BlockCid cid.Cid
+	MsgCid   cid.Cid
+}
+
+// MsgSearchFn is the function signature used to find a message
+type MsgSearchFn func(context.Context, *fast.Filecoin, *types.SignedMessage) (bool, error)
+
+// WaitForChainMessage iterates over the chain until the provided function `fn` returns true.
+func WaitForChainMessage(ctx context.Context, node *fast.Filecoin, fn MsgSearchFn) (*MsgInfo, error) {
+	for {
+		select {
+		case <-ctx.Done():
+			return nil, ctx.Err()
+		case <-CtxSleepDelay(ctx):
+			dec, err := node.ChainLs(ctx)
```

Comment:
This inside a for loop is potentially a really expensive way to listen for incoming blocks. It would be good to either use a block topic subscriber or track known blocks to prevent the next loop from iterating all the way back to genesis each loop.

Reply:
I'm guessing @travisperson really wants something like https://github.com/filecoin-project/go-filecoin/issues/3230 to replace this whole construction.